projects/core/src/product/services/product-loading.service.ts

```diff
@@ -46,6 +46,9 @@ export class ProductLoadingService {
   ) {}
 
   get(productCode: string, scopes: string[]): Observable<Product> {
+    if (this.isInvalidProductCode(productCode)) {
```

Comment:
I recommend creating a utility function for this and using it in areas where prototype pollution might occur

Reply:
ok, I see a PR was created here: 
https://github.com/SAP/spartacus/pull/19432/files